src/httpClient.ts

```diff
@@ -270,3 +276,35 @@ function make_request_timing(
       deserialize_end && body_read_end ? deserialize_end - body_read_end : 0,
   };
 }
+
+function convertHeadersType(
+  headers: Record<string, string | string[] | undefined>,
+): Record<string, string> {
+  for (const key in headers) {
+    const v = headers[key];
+    if (v === undefined) {
+      delete headers[key];
+    } else if (Array.isArray(v)) {
+      headers[key] = v[0];
+    }
+  }
+  return headers as Record<string, string>;
+}
+
+async function consumeResponseText(response: Dispatcher.ResponseData): Promise<{
+  body_text: string;
+  body_read_end: number;
+}> {
+  if (response.headers["content-encoding"] == "gzip") {
+    let body_buffer = await response.body.arrayBuffer();
+    let body_read_end = performance.now();
+
+    let gunzip_buffer = await gunzip_async(body_buffer);
```

Comment:
> stream gunzip?

I feel like this is not worth it over the benefits of one-shot, especially the timing we can add here

>  separate metric for gunzip / json?

yes; you can add a combined if you prefer, but I feel like that's just body_read

Reply:
we already have deserialize_time which is the combined time, question is whether to make multiple timings for the two steps

Reply:
Yes I definitely think we should